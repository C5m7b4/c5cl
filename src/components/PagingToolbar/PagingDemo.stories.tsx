import React from 'react';
import { Story, Meta } from '@storybook/react';

import PagingDemo from './PagingDemo';

export default {
  title: 'Widgets/PagingDemo',
  component: PagingDemo,
} as Meta;

const Template: Story = (args) => <PagingDemo />;

export const PagingDemoExample = Template.bind({});
