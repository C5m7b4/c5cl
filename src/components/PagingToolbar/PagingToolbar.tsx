import React, { useState, useEffect } from 'react';

import Tooltip from '../Tooltip';
import { Icons } from '../Icons';

import './PagingToolbar.css';

export interface PagingToolbarProps<T> {
  data: T[];
  totalRecords: number;
  recordsPerPage?: number;
  onChange?: (start: number, end: number) => void;
  onRecordsPerPageChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const initialPagingToolbarState = {
  data: [],
  recordsPerPage: 50,
  totalPages: 0,
  totalRecords: 0,
  start: 0,
  end: 0,
};

const style = { width: '32px', height: '32px' };

function PagingToolbar<T>(props: PagingToolbarProps<T>) {
  const [state, setState] = useState(initialPagingToolbarState);
  const [currentPage, setCurrentPage] = useState(0);

  const { data, totalRecords, recordsPerPage = state.recordsPerPage } = props;

  const {
    previous: Previous,
    next: Next,
    backward: Backward,
    forward: Forward,
  } = Icons;

  useEffect(() => {
    let totalPages = Math.ceil(totalRecords / recordsPerPage);
    if (totalPages === 0) {
      totalPages = 1;
    }

    setState({
      ...state,
      totalPages,
      recordsPerPage,
      end: state.end === 0 ? recordsPerPage : state.end,
    });
  }, [data, totalRecords, props.recordsPerPage]);

  const handleNext = () => {
    /* istanbul ignore else */
    if (props.onChange) {
      let start = state.start + recordsPerPage;
      let end = state.end + recordsPerPage;
      if (end > totalRecords) {
        props.onChange(start, totalRecords);
      } else {
        props.onChange(start, end);
      }

      setState({
        ...state,
        start,
        end,
      });
    }

    setCurrentPage(currentPage + 1);
  };

  const handlePrev = () => {
    /* istanbul ignore else */
    if (props.onChange) {
      let start = state.start - recordsPerPage;
      let end = state.end - recordsPerPage;
      props.onChange(start, end);
      setState({
        ...state,
        start,
        end,
      });
    }
    setCurrentPage(currentPage - 1);
  };

  const handleFirst = () => {
    /* istanbul ignore else */
    if (props.onChange) {
      props.onChange(0, recordsPerPage);
    }
    setCurrentPage(0);
    setState({
      ...state,
      start: 0,
      end: recordsPerPage,
    });
  };

  const handleLast = () => {
    /* istanbul ignore else */
    if (props.onChange) {
      props.onChange(totalRecords - recordsPerPage, totalRecords);
    }
    const lastPage = Math.ceil(totalRecords / recordsPerPage) - 1;
    setCurrentPage(lastPage);
    setState({
      ...state,
      start: totalRecords - recordsPerPage,
      end: totalRecords,
    });
  };

  const calculateRemainingRecords = () => {
    return totalRecords;
  };

  const handleRecordsPerPageChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ) => {
    /* istanbul ignore else */
    if (props.onRecordsPerPageChange) {
      props.onRecordsPerPageChange(e);
    }
  };

  const handleCurrentPageChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const cp = +e.target.value - 1;
    setCurrentPage(cp);
    let start = recordsPerPage * cp;
    let end = start + recordsPerPage;
    /* istanbul ignore else */
    if (props.onChange) {
      props.onChange(start, end);
    }
    setState({
      ...state,
      start,
      end,
    });
  };

  return (
    <div className="paging-toolbar-wrapper">
      <div style={{ width: '100%', display: 'flex' }}>
        <div style={{ margin: '0 auto' }}>
          {currentPage === 0 ? (
            <React.Fragment>
              <span>
                <Previous
                  theme="dark"
                  type="dark"
                  style={{ ...style, opacity: '0.2' }}
                />
              </span>
              <span>
                <Backward
                  theme="dark"
                  type="dark"
                  style={{ ...style, opacity: '0.2' }}
                />
              </span>
              <span
                style={{
                  marginLeft: '10px',
                  marginRight: '10px',
                  height: '8px',
                }}
              ></span>
            </React.Fragment>
          ) : (
            <React.Fragment>
              <Tooltip message={'First Page'} position="bottom" theme="dark">
                <span
                  className="paging-button"
                  data-testid="handle-first"
                  onClick={handleFirst}
                >
                  <Previous theme="dark" type="dark" style={style} />
                </span>
              </Tooltip>
              <Tooltip message={'Previous Page'} position="bottom" theme="dark">
                <span
                  style={{ marginLeft: '10px', marginRight: '10px' }}
                  className="paging-button"
                  onClick={handlePrev}
                  data-testid="handle-prev"
                >
                  <Backward theme="dark" type="dark" style={style} />
                </span>
              </Tooltip>
            </React.Fragment>
          )}
          <span style={{ verticalAlign: 'text-bottom' }}>
            <span>Page</span>
            <input
              type="number"
              className="paging-toolbar-current-page"
              value={currentPage + 1}
              onChange={(e) => handleCurrentPageChange(e)}
              data-testid="current-page-input"
              min="0"
              max={Math.ceil(totalRecords / recordsPerPage)}
            />
            <span> of {state.totalPages}</span>
          </span>

          {currentPage == state.totalPages - 1 ? (
            <React.Fragment>
              <span style={{ marginLeft: '10px', marginRight: '10px' }}>
                <Next
                  theme="dark"
                  type="dark"
                  style={{ ...style, opacity: '0.2' }}
                />
              </span>
              <span>
                <Forward
                  theme="dark"
                  type="dark"
                  style={{ ...style, opacity: '0.2' }}
                />
              </span>
            </React.Fragment>
          ) : (
            <React.Fragment>
              <Tooltip message="Next Page" position="bottom" theme="dark">
                <span
                  style={{ marginLeft: '10px', marginRight: '10px' }}
                  className="paging-button"
                  data-testid="handle-next"
                  onClick={handleNext}
                >
                  <Forward theme="dark" type="dark" style={style} />
                </span>
              </Tooltip>
              <Tooltip message="Last Page" position="bottom" theme="dark">
                <span
                  className="paging-button"
                  data-testid="handle-last"
                  onClick={handleLast}
                >
                  <Next theme="dark" type="dark" style={style} />
                </span>
              </Tooltip>
            </React.Fragment>
          )}
        </div>
      </div>
      <div style={{ width: '100%', textAlign: 'center', marginTop: '10px' }}>
        <div style={{ margin: '-8px auto' }}>
          <span>Show</span>
          <input
            type="number"
            className="paging-records-per-page"
            value={recordsPerPage}
            data-testid="rpp-input"
            onChange={(e) => handleRecordsPerPageChange(e)}
          />
          <span>Records per Page</span>
          <span className="paging-summary">
            <span>{`Displaying ${state.start} - ${
              state.end > totalRecords ? calculateRemainingRecords() : state.end
            } of ${totalRecords}`}</span>
          </span>
        </div>
      </div>
    </div>
  );
}

export default PagingToolbar;
