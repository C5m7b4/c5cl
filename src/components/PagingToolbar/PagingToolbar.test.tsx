import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import PagingToolbar from './PagingToolbar';

const testData = [
  {
    id: 1,
    storeName: 'store1',
  },
  {
    id: 2,
    storeName: 'store2',
  },
];

const testFn = jest.fn();
const testFn1 = jest.fn();

describe('Paging Toolbar', () => {
  test('should render properly', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={1}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle zero results', () => {
    const { container } = render(
      <PagingToolbar
        data={[]}
        currentPage={1}
        totalRecords={0}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle next button click', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={1}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    const btn = screen.getByTestId('handle-next');
    fireEvent.click(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle last button click', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={1}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    const btn = screen.getByTestId('handle-last');
    fireEvent.click(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle first button click', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={1}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    const btn = screen.getByTestId('handle-first');
    fireEvent.click(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle prev button click', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={1}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    const btn = screen.getByTestId('handle-prev');
    fireEvent.click(btn);
    expect(container).toMatchSnapshot();
  });
  test('should handle last page', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={4}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={25}
        end={50}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle current page change', () => {
    const { container } = render(
      <PagingToolbar
        data={testData}
        currentPage={1}
        totalRecords={100}
        handlePageChange={testFn}
        recordsPerPage={20}
        handleRecordsPerPageChange={testFn1}
        start={0}
        end={25}
      />
    );
    userEvent.type(screen.getByTestId('current-page-input'), '2');
  });
});
