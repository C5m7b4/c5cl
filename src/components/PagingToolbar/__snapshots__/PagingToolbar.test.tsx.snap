// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging Toolbar should handle first button click 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="2"
          />
          <span>
             of 
            5
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-next"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-last"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 0 - 25 of 100
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paging Toolbar should handle last button click 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="2"
          />
          <span>
             of 
            5
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-next"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-last"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 0 - 25 of 100
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paging Toolbar should handle last page 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="5"
          />
          <span>
             of 
            5
          </span>
        </span>
        <span
          style="margin-left: 10px; margin-right: 10px;"
        >
          <svg
            fill-opacity="var(--toastify__toast-icon-color-dark)"
            height="100%"
            style="width: 32px; height: 32px;"
            viewBox="0 0 32 32"
            width="100%"
          >
            <path
              d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
            />
          </svg>
        </span>
        <span>
          <svg
            fill-opacity="var(--toastify__toast-icon-color-dark)"
            height="100%"
            style="width: 32px; height: 32px;"
            viewBox="0 0 32 32"
            width="100%"
          >
            <path
              d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 25 - 50 of 100
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paging Toolbar should handle next button click 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="2"
          />
          <span>
             of 
            5
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-next"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-last"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 0 - 25 of 100
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paging Toolbar should handle prev button click 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="2"
          />
          <span>
             of 
            5
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-next"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-last"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 0 - 25 of 100
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paging Toolbar should handle zero results 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="2"
          />
          <span>
             of 
            1
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-next"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-last"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 0 - 0 of 0
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Paging Toolbar should render properly 1`] = `
<div>
  <div
    class="paging-toolbar-wrapper"
  >
    <div
      style="width: 100%; display: flex;"
    >
      <div
        style="margin: 0px auto;"
      >
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-first"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M8 28v-24h4v11l10-10v22l-10-10v11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-prev"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M18 5v10l10-10v22l-10-10v10l-11-11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          style="vertical-align: text-bottom;"
        >
          <span>
            Page
          </span>
          <input
            class="paging-toolbar-current-page"
            data-testid="current-page-input"
            type="number"
            value="2"
          />
          <span>
             of 
            5
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-next"
              style="margin-left: 10px; margin-right: 10px;"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M16 27v-10l-10 10v-22l10 10v-10l11 11z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="mcl-tooltip-wrapper"
        >
          <span
            class="mcl-tooltip-trigger"
          >
            <span
              class="paging-button"
              data-testid="handle-last"
            >
              <svg
                fill-opacity="var(--toastify__toast-icon-color-dark)"
                height="100%"
                style="width: 32px; height: 32px;"
                viewBox="0 0 32 32"
                width="100%"
              >
                <path
                  d="M24 4v24h-4v-11l-10 10v-22l10 10v-11z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      style="width: 100%; text-align: center; margin-top: 10px;"
    >
      <div
        style="margin: -8px auto;"
      >
        <span>
          Show
        </span>
        <input
          class="paging-records-per-page"
          type="number"
          value="20"
        />
        <span>
          Records per Page
        </span>
        <span
          class="paging-summary"
        >
          <span>
            Displaying 0 - 25 of 100
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
